oracle/pkg/store/store.go

```diff
@@ -157,6 +172,11 @@ func (s *Store) AddEncryptedCommitment(
 	commitmentSignature []byte,
 	dispatchTimestamp uint64,
 ) error {
+	// check if the commitmentIdx already exists
+	if s.encryptedCommitmentExists(ctx, commitmentIdx) {
```

Comment:
“Wouldn’t it be better to add error handling (e.g., skip if duplicated) instead of checking for the existence of the commitment for every request?”